Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm

```diff
@@ -455,34 +455,20 @@ - (void)_updateScreenTimeViewGeometry
     [[_screenTimeWebpageController view] setFrame:bounds];
 }
 
-- (void)_updateScreenTimeShieldVisibilityForWindow
+- (void)_updateScreenTimeBasedOnWindowVisibility
 {
     BOOL viewIsInWindow = !!self.window;
-
-    BOOL showsSystemScreenTimeBlockingView = [_configuration _showsSystemScreenTimeBlockingView];
-
-    if (viewIsInWindow) {
-        if (!showsSystemScreenTimeBlockingView && _screenTimeBlurredSnapshot) {
-            [_screenTimeBlurredSnapshot setHidden:NO];
-            RELEASE_LOG(ScreenTime, "Screen Time has updated visibility to show blurred view.");
-        } else if (showsSystemScreenTimeBlockingView) {
-            [[_screenTimeWebpageController view] setHidden:NO];
-            RELEASE_LOG(ScreenTime, "Screen Time has updated visibility to show system shield.");
-        }
-    } else {
-        if (_screenTimeBlurredSnapshot) {
-            [_screenTimeBlurredSnapshot setHidden:YES];
-            RELEASE_LOG(ScreenTime, "Screen Time has updated visibility to hide blurred view.");
-        } else if (showsSystemScreenTimeBlockingView) {
-            [[_screenTimeWebpageController view] setHidden:YES];
-            RELEASE_LOG(ScreenTime, "Screen Time has updated visibility to hide system shield.");
-        }
-    }
-
     BOOL viewIsVisible = viewIsInWindow;
+
 #if PLATFORM(MAC)
     viewIsVisible &= ((self.window.occlusionState & NSWindowOcclusionStateVisible) == NSWindowOcclusionStateVisible);
```

Comment:
FYI this is doing a bitwise AND operation. Better to avoid the ambiguity because I know you want logical AND instead.

```suggestion
    viewIsVisible = viewIsVisible && ((self.window.occlusionState & NSWindowOcclusionStateVisible) == NSWindowOcclusionStateVisible);
```